src/Builder/Encoder/VariableEncoder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder\Encoder;
+
+use MongoDB\Builder\Expression\Variable;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+
+/** @template-extends AbstractExpressionEncoder<string, Variable> */
+class VariableEncoder extends AbstractExpressionEncoder
+{
+    /** @template-use EncodeIfSupported<string, Variable> */
+    use EncodeIfSupported;
+
+    public function canEncode(mixed $value): bool
+    {
+        return $value instanceof Variable;
+    }
+
+    public function encode(mixed $value): string
+    {
+        if (! $this->canEncode($value)) {
+            throw UnsupportedValueException::invalidEncodableValue($value);
+        }
+
+        return '$$' . $value->name;
```

Comment:
In FieldPathEncoder you have the following:

```
// TODO: needs method because of interface
```

Is this also affected? If so, can we cover both in a single JIRA ticket?

Reply:
@alcaeus: :point_up:

Reply:
Tracked in PHPLIB-1379.